### PR TITLE
Suggestion rsa_import_hex

### PR DESCRIPTION
**Pull request by sjaeckel**
I would like to propose adding the following function for importing RSA key from a set of numbers represented as hexadecimal strings:

```
int rsa_import_hex(char *N, char *e, char *d, char *p, char *q,
                            char *dP, char *dQ, char *qP, rsa_key *key)
{
   int err;

   LTC_ARGCHK(key         != NULL);
   LTC_ARGCHK(N           != NULL);
   LTC_ARGCHK(e           != NULL);
   LTC_ARGCHK(ltc_mp.name != NULL);

   err = mp_init_multi(&key->e, &key->d, &key->N, &key->dQ, 
                                &key->dP, &key->qP, &key->p, &key->q, NULL);
   if (err != CRYPT_OK) return err;

   if ((err = mp_read_radix(key->N , N , 16)) != CRYPT_OK)   { goto LBL_ERR; }
   if ((err = mp_read_radix(key->e , e , 16)) != CRYPT_OK)   { goto LBL_ERR; }
   if (d && p && q && dP && dQ && qP && strlen(d)>0 && strlen(p)>0 && 
       strlen(q)>0 && strlen(dP)>0 && strlen(dQ)>0 && strlen(qP)>0) {
     if ((err = mp_read_radix(key->d , d , 16)) != CRYPT_OK) { goto LBL_ERR; }
     if ((err = mp_read_radix(key->p , p , 16)) != CRYPT_OK) { goto LBL_ERR; }
     if ((err = mp_read_radix(key->q , q , 16)) != CRYPT_OK) { goto LBL_ERR; }
     if ((err = mp_read_radix(key->dP, dP, 16)) != CRYPT_OK) { goto LBL_ERR; }
     if ((err = mp_read_radix(key->dQ, dQ, 16)) != CRYPT_OK) { goto LBL_ERR; }
     if ((err = mp_read_radix(key->qP, qP, 16)) != CRYPT_OK) { goto LBL_ERR; }
     key->type = PK_PRIVATE;
   }
   else {
     key->type = PK_PUBLIC;
   }
   return CRYPT_OK;

LBL_ERR:
   mp_clear_multi(key->d, key->e, key->N, key->dQ, key->dP, key->qP, key->p, key->q, NULL);
   return err;
}
```

### CODE REVIEW

**Comment by sjaeckel** [comment]
We should also have a testcase for this

**Comment by karel-m** [comment]
> We should also have a testcase for this

I'll add some to rsa_test.c (+ dsa_test.c)

**Comment by karel-m** [comment]
after merging & closing this PR do not forget to close #136 as well

File: src/pk/rsa/rsa_import_radix.c [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+   LTC_ARGCHK(N           != NULL);
+   LTC_ARGCHK(e           != NULL);
+   LTC_ARGCHK(ltc_mp.name != NULL);
+
+   err = mp_init_multi(&key->e, &key->d, &key->N, &key->dQ, &key->dP, &key->qP, &key->p, &key->q, NULL);
+   if (err != CRYPT_OK) return err;
+
+   if ((err = mp_read_radix(key->N , N , radix)) != CRYPT_OK)    { goto LBL_ERR; }
+   if ((err = mp_read_radix(key->e , e , radix)) != CRYPT_OK)    { goto LBL_ERR; }
+   if (d && p && q && dP && dQ && qP && strlen(d)>0 && strlen(p)>0 &&
```
Comment (sjaeckel) [inline src/pk/rsa/rsa_import_radix.c]: a private key doesn't necessarily have CRT parameters!

**Pull request merged**